### PR TITLE
[OPIK-3918] [FE] Project view: Add experiments tab to project view

### PR DESCRIPTION
**Pull request by JetoPistola**
## Details

https://github.com/user-attachments/assets/79c488c0-0ced-415a-95d6-0a5ecd1fa306

image

image

image

This PR implements a new **Experiments tab** within the project view, allowing users to view and manage all experiments related to a specific project. The implementation follows the existing patterns from the main Experiments page and Prompt Experiments tab.

### Key Features Implemented:

- **New Experiments Tab**: Added a dedicated tab in the project view (positioned after Annotation queues) to display project-specific experiments
- **Project Context Filtering**: Automatically filters experiments by the current project ID, eliminating the need for manual project selection
- **Reusable Components**: Leverages existing experiment table infrastructure (`useGroupedExperimentsList`, `useExperimentsTableConfig`) with project-specific adaptations
- **Filter & Group Columns**: Excludes project column from filters and grouping options since users are already in a project context
- **Explainer Callout**: Added contextual help explaining what experiments are, consistent with other tabs
- **Create Experiment Button**: Integrated "Create new experiment" button that pre-fills the project name in the generated Python code
- **Empty State**: Displays helpful message and clickable "Create new experiment" link when no experiments exist
- **Refresh Button**: Added refresh functionality to reload experiments list
- **Proper Hook Integration**: Modified `useGroupedExperimentsList` to correctly handle `projectId` parameter when grouping by other fields

### Technical Implementation:

1. **New Component**: `ProjectExperimentsTab.tsx` - Main component for the experiments tab within project view
2. **Hook Modifications**:
   - `useGroupedExperimentsList`: Enhanced to prioritize directly provided `projectId` over filter-extracted values
   - `useExperimentsGroupsAndFilters`: Added `projectId` parameter support for metadata path autocomplete
3. **Dialog Enhancement**: `AddExperimentDialog` now accepts optional `projectName` to pre-fill in generated code
4. **Shared Constants**: Reuses `FILTER_AND_GROUP_COLUMNS` from shared utilities, filtering out project column

### UI/UX Improvements:

- Consistent with existing Experiments page patterns (layout, buttons, empty states)
- Matches Figma design specifications for empty state text
- Provides clear visual feedback and helpful actions for users
- Maintains responsive design and accessibility standards

## Change checklist

- [x] User facing
- [x] Documentation update

## Issues

- OPIK-3918

### Manual Testing Steps:

1. Navigate to any project view
2. Click on the "Experiments" tab (after Annotation queues)
3. Verify experiments are filtered to the current project
4. Test filtering and grouping by dataset, tags, and metadata
5. Verify "Create new experiment" button opens dialog with project pre-filled
6. Test empty state when no experiments exist
7. Verify refresh button reloads the experiments list
8. Test pagination and sorting functionality

### Edge Cases Tested:

- Projects with no experiments (empty state)
- Projects with many experiments (pagination)
- Grouping by dataset while in project context
- Filtering experiments by various criteria

## Documentation

N/A - UI feature, no configuration changes required

### CODE REVIEW

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-4905.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-4905
- **Version:** 1.9.99-4905-merge-1044
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-4905%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+
+    if (isGroupingByDataset) {
+      filteredColumns = filteredColumns.filter(
+        (col) => col.id !== COLUMN_DATASET_ID,
+      );
+    }
+
+    return filteredColumns;
+  }, [groups, columnsDef]);
+
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:429]: Doesn't it work for datasets? 
I think we can display columns for datasets as weil now, or extend this logic so it doesn't allow users to pick tags or configuration if they are selected for grouping

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+
+  const [sortedColumns, setSortedColumns] = useLocalStorageState<ColumnSort[]>(
+    COLUMNS_SORT_KEY,
+    {
+      defaultValue: [],
+    },
+  );
+
+  const columnsDef: ColumnData<GroupedExperiment>[] = useMemo(() => {
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:155]: maybe, it's worth reusing the same columns as with an experiment page

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx (hunk 1 above)
Comment (JetoPistola) [reply apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:429]: @aadereiko That's a good note, regarding tags and configuration

The columns are also visible when grouped by, as mentioned in https://github.com/comet-ml/opik/pull/4905#discussion_r2726524252, 

Created 2 tickets:

OPIK-4029: [FE] Extend availableColumns logic to filter tags and configuration when grouped
OPIK-4030: [FE] Auto-hide grouped columns and restore visibility state on ungroup

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-4905.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-4905
- **Version:** 1.9.99-4905-merge-1053
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-4905%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+    let filteredColumns = columnsDef;
+
+    if (isGroupingByDataset) {
+      filteredColumns = filteredColumns.filter(
+        (col) => col.id !== COLUMN_DATASET_ID,
+      );
+    }
+
+    return filteredColumns;
+  }, [groups, columnsDef]);
```
[the remark this answers is not in the document]
Comment (JetoPistola) [reply apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:428]: Hey Baz, let's skip with this PR

Will be dealt with, in future ticket OPIK-4030 - [FE] Auto-hide grouped columns and restore visibility state on ungroup, so then selected columns like dataset will remain selected, but will be hidden as a regular column in the table once grouped by

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+import PageBodyStickyContainer from "@/components/layout/PageBodyStickyContainer/PageBodyStickyContainer";
+import PageBodyStickyTableWrapper from "@/components/layout/PageBodyStickyTableWrapper/PageBodyStickyTableWrapper";
+import DataTablePagination from "@/components/shared/DataTablePagination/DataTablePagination";
+import { useIsFeatureEnabled } from "@/components/feature-toggles-provider";
+import { FeatureToggleKeys } from "@/types/feature-toggles";
+import ExplainerCallout from "@/components/shared/ExplainerCallout/ExplainerCallout";
+
+const STORAGE_KEY_PREFIX = "project-experiments";
+const PAGINATION_SIZE_KEY = "project-experiments-pagination-size";
+const COLUMNS_SORT_KEY = "project-experiments-columns-sort";
```
[the remark this answers is not in the document]
Comment (JetoPistola) [reply apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:80]: Looks like a potential tech debt, and for all pages, in Opik

Created a ticket, OPIK-4031, to decide later if we wish to do the STORAGE_KEY_PREFIX refactor in all pages & tabs, logic somewhat repeats, and there are good candidates on solution

For this PR, let's keep the solution for consistency Baz

File: apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx (hunk 2 above)
Comment (JetoPistola) [reply apps/opik-frontend/src/components/pages/ProjectPage/ProjectExperimentsTab/ProjectExperimentsTab.tsx:155]: That's a good question @aadereiko , followed as consistency

1 SOT column (or columns) to be used in all places like the page and tabs in other resource pages

There is also the page/tab similarity in 90-95%, where it's DRY vs SRP (It's where there is no 1 right solution as to going with DRY or SRP, and went with SRP as it's follows the rest of the code, it should be a team decision)

Created ticket OPIK-4033

**Comment by JetoPistola** [comment]
We decided to close the PR for time being, will iterate on it in the future.
